### ISSUE TITLE
Multithreaded configuration of packages

### ISSUE THREAD

**Issue by Tremoneck** [microsoft/vcpkg#20326]
I recently had to install a lot of packages. From the output of vcpkg it seems that the packages are sequentially downloaded, patched, configured, build and then installed. By looking at the Task-Manager it looked like all, except the building stage, were happening only a single thread and not using all CPU resources. 

There might be a chance to speed up installing of big packages by a significant factor by running several non-building tasks together with a single building task, without slowing the building task down significantly.

**Comment by PhoebeHui** [comment]
Right, AFAIK, some ports that configured with cmake aleady configured the project in parallel, however, some of these project may be disabled parallel configure, this is needed for libraries which write back into their source directory during configure.

Closing in favor of https://github.com/microsoft/vcpkg/discussions/19066.
